### DISCUSSION TITLE
Undocumented Web API response change in Result.case_counts length

### DISCUSSION THREAD

**Discussion by Aldaviva** [testspace-com/support#29] [Q&A] [answered]
Hello,

The length of the `case_counts` array in the `Result` response object recently increased from 4 to 5. This change was deployed some time between 2022-10-09 and 2022-10-16. One effect of this change is that [Shields.io](https://shields.io/category/test-results) badges for Testspace all return "invalid response data" now, as they have a [validator for max length 4](https://github.com/badges/shields/blob/6dc8aac4513aeae30a56c92ecfd12f37fc0b7cff/services/testspace/testspace-base.js#L16) on this property value, which hasn't changed recently.

![Shields.io Testspace sample space](https://img.shields.io/testspace/tests/swellaby/swellaby:testspace-sample/main)

The current official [List results](https://help.testspace.com/reference/web-api#list-results) API method documentation still describes the `case_counts` property of the [Result](https://help.testspace.com/reference/web-api#results) response object as

|Field|Type|Notes|
|---|---|---|
|`case_counts`|array|The contained cases [passed, failed, na, errored] counters|

However, an actual API request and response now looks like
```http
GET /api/projects/testspace-com%3Ademo/spaces/main/results HTTP/1.1
Host: demo.testspace.com
```
```js
[
  {
    "case_counts": [64, 9, 2, 0, 0],
    // ...
  },
  // ...
]
```

I checked the official [Changelog](https://help.testspace.com/changelog) and did not see any changes that were recent or that referred to the Response object.

---

- What is the new meaning of the `case_counts` array?
    I'm guessing some new value was appended to the end of the array, but all of the spaces I found, including some of my own, always have 0 in the last two positions, so I'm not sure where the new value was inserted, or what it means.
- Can the [official documentation](https://help.testspace.com/reference/web-api#results) be updated to reflect these API changes?

Thanks!

**Comment by ivailop** [comment, marked as answer]
Hi @Aldaviva,

Thanks for reporting the issue. As the Testspace badges at [Shilds.io](https://shields.io/category/test-results) are not officially supported by us we were not aware of the problem.

Yes, there was a new Testspace release deployed last week that added a 5th number to the `case_counts` array. It is meant to capture "untested" count, which is meaningful only for Manual testing. 

We are sorry we are behind with updating our documentation about that change.

Would you like to submit a fix to https://github.com/badges/shields or you prefer us to follow up with that? Notice, some of the comments in there may have to be also adjusted as they point to non-existing help-urls.

**Reply by Aldaviva** [reply]
Thanks! I can make a first attempt at a pull request and let you know if there are any questions we need help with. I'll update the help URLs too.

**Reply by Aldaviva** [reply]
Filed https://github.com/badges/shields/pull/8544.

**Reply by Aldaviva** [reply]
PR approved and merged.

**Reply by ivailop** [reply]
Thank you @Aldaviva!
We also updated our help - https://help.testspace.com/reference/web-api#results

**Reply by Aldaviva** [reply]
Looking good. Shields.io have pushed a release to production and it seems to be fully resolved now.

[![Testspace](https://img.shields.io/testspace/tests/Aldaviva/Aldaviva:PowerOverInternet/master?passed_label=passing&failed_label=failing&logo=data%3Aimage%2Fsvg%2Bxml%3Bbase64%2CPHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCA4NTkgODYxIj48cGF0aCBkPSJtNTk4IDUxMy05NCA5NCAyOCAyNyA5NC05NC0yOC0yN3pNMzA2IDIyNmwtOTQgOTQgMjggMjggOTQtOTQtMjgtMjh6bS00NiAyODctMjcgMjcgOTQgOTQgMjctMjctOTQtOTR6bTI5My0yODctMjcgMjggOTQgOTQgMjctMjgtOTQtOTR6TTQzMiA4NjFjNDEuMzMgMCA3Ni44My0xNC42NyAxMDYuNS00NFM1ODMgNzUyIDU4MyA3MTBjMC00MS4zMy0xNC44My03Ni44My00NC41LTEwNi41UzQ3My4zMyA1NTkgNDMyIDU1OWMtNDIgMC03Ny42NyAxNC44My0xMDcgNDQuNXMtNDQgNjUuMTctNDQgMTA2LjVjMCA0MiAxNC42NyA3Ny42NyA0NCAxMDdzNjUgNDQgMTA3IDQ0em0wLTU1OWM0MS4zMyAwIDc2LjgzLTE0LjgzIDEwNi41LTQ0LjVTNTgzIDE5Mi4zMyA1ODMgMTUxYzAtNDItMTQuODMtNzcuNjctNDQuNS0xMDdTNDczLjMzIDAgNDMyIDBjLTQyIDAtNzcuNjcgMTQuNjctMTA3IDQ0cy00NCA2NS00NCAxMDdjMCA0MS4zMyAxNC42NyA3Ni44MyA0NCAxMDYuNVMzOTAgMzAyIDQzMiAzMDJ6bTI3NiAyODJjNDIgMCA3Ny42Ny0xNC44MyAxMDctNDQuNXM0NC02NS4xNyA0NC0xMDYuNWMwLTQyLTE0LjY3LTc3LjY3LTQ0LTEwN3MtNjUtNDQtMTA3LTQ0Yy00MS4zMyAwLTc2LjY3IDE0LjY3LTEwNiA0NHMtNDQgNjUtNDQgMTA3YzAgNDEuMzMgMTQuNjcgNzYuODMgNDQgMTA2LjVTNjY2LjY3IDU4NCA3MDggNTg0em0tNTU3IDBjNDIgMCA3Ny42Ny0xNC44MyAxMDctNDQuNXM0NC02NS4xNyA0NC0xMDYuNWMwLTQyLTE0LjY3LTc3LjY3LTQ0LTEwN3MtNjUtNDQtMTA3LTQ0Yy00MS4zMyAwLTc2LjgzIDE0LjY3LTEwNi41IDQ0UzAgMzkxIDAgNDMzYzAgNDEuMzMgMTQuODMgNzYuODMgNDQuNSAxMDYuNVMxMDkuNjcgNTg0IDE1MSA1ODR6IiBmaWxsPSIjZmZmIi8%2BPC9zdmc%2B)](https://aldaviva.testspace.com/spaces/192613)

Thanks for all your help with this, @ivailop!
